### PR TITLE
vm: update tests to v13 (cancun tests)

### PR DESCRIPTION
**Pull request by jochem-brouwer**
Updates to https://github.com/ethereum/tests/releases/tag/v13

Some gotcha's:

Test data had to be updated to ensure that the RLP-encoded 4844 txs have a `to` field. Some test data has `to` field with empty bytes, this is not possible per spec.

The `stateRoot` of some tests updates because we implicitly create the beacon root address. If there is no code there (if there is storage there it does not matter) it should be deleted anyways. The change in `runBlock.ts` changes the state root in case that the beacon root address has no code (it is now wiped from the Trie instead of creating a semi-empty account (it has storage, but per definition if it is touched should be deleted after a block run).

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
On hold, blockchain tests do seem to ignore EIP 4788.

**Comment by jochem-brouwer** [comment]
Never mind, it's the touched logic again. Should pass now, have included some 4844 spec fixes.

**Comment by jochem-brouwer** [comment]
Nvm, the blockchain/state tests pass locally but it seems that tx now errors out

File: packages/tx/src/constants.ts
Comment (g11tech) [inline packages/tx/src/constants.ts:6]: i think we should move this constant to the teser/runner since this would be hf dependant anyway

**Review by g11tech — COMMENTED** [review]
mostly looks good to me can approve once CI passes

File: packages/tx/src/constants.ts
Comment (jochem-brouwer) [reply packages/tx/src/constants.ts:6]: We can move `MAX_BLOB_GAS_PER_BLOB` and `GAS_PER_BLOB` into common?

**Comment by jochem-brouwer** [comment]
Updated the opening comment on this PR in case there is some confusion why some test files had to be updated (explicitly checked all)

File: packages/tx/src/constants.ts
Comment (jochem-brouwer) [reply packages/tx/src/constants.ts:6]: (and then derive max amount of blobs from that)

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**
